### DISCUSSION TITLE
access repl

### DISCUSSION THREAD

**Discussion by majidaldo** [quarto-dev/quarto-cli#5664] [Q&A] [answered]
i'm really liking quarto. i'm primarily using jupyter for execution.

what bothers me is that sometimes i just need to access the repl. how do i do that ??

quarto is nice for iterating on _rendering_ while the repl is good for debugging and code dev.

**Comment by mcanouil** [comment]
It's unclear what's your need ...
REPL is not a specific thing and has nothing to do with Quarto CLI.

Use the usual REPL for whatever language you are using in a terminal within an IDE or not.

**Reply by majidaldo** [reply]
> It's unclear what's your need ... REPL is not a specific thing and has nothing to do with Quarto CLI.
> 
> Use the usual REPL for whatever language you are using in a terminal within an IDE or not.

Quarto has access to the kernel state .  If there's an error I just get the trace. I don't get to step through it to debug like I can in a repl.

**Reply by mcanouil** [reply, marked as answer]
If you are writing a flat file (qmd), I don't think there is a "state" saved.
You can execute code cells in REPL.

If you are using a Jupyter Notebook, then you have access to all the features from it and in the end you can render it using Quarto.

For example, using `knitr`:
<img width="1624" alt="Screenshot 2023-05-24 at 21 11 01" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/1186a4f9-f94a-41c6-b978-a533fab5e4e0">

Using `Jupyter`:
<img width="1624" alt="Screenshot 2023-05-24 at 21 18 51" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/1746f0ad-c175-496e-ab2b-df58d28e0e13">

**Reply by majidaldo** [reply]
> If you are writing a flat file (qmd), I don't think there is a "state" saved. You can execute code cells in REPL.
> 
> If you are using a Jupyter Notebook, then you have access to all the features from it and in the end you can render it using Quarto.
> 
> For example, using `knitr`: <img alt="Screenshot 2023-05-24 at 21 11 01" width="1624" src="https://user-images.githubusercontent.com/8896044/240718498-1186a4f9-f94a-41c6-b978-a533fab5e4e0.png">
> 
> Using `Jupyter`: <img alt="Screenshot 2023-05-24 at 21 18 51" width="1624" src="https://user-images.githubusercontent.com/8896044/240719769-1746f0ad-c175-496e-ab2b-df58d28e0e13.png">

Thanks but to me this means there is no repl access. I don't want two things executing on my document. Furthermore, the jupyter Run Cell execution is not like terminal interaction.

Technically, quarto could intercept stack traces and drop the user in them. If it behaves like jupyter server then https://jupyter-client.readthedocs.io/en/stable/kernels.html#connection-files. Also quarto could be given the kernel connection info instead of the kernel name.

**Reply by mcanouil** [reply]
Feel free to open a feature request discussion and explaining/describing the best you can what you would want to be implemented and eventually hints on how to do it.

https://github.com/quarto-dev/quarto-cli/discussions/categories/feature-requests
